src/panels/lovelace/cards/hui-picture-elements-card.ts

```diff
@@ -0,0 +1,103 @@
+import { html, LitElement } from "@polymer/lit-element";
+
+import createHuiElement from "../common/create-hui-element.js";
+
+import { LovelaceCard, LovelaceConfig } from "../types";
+import { HomeAssistant } from "../../../types.js";
+import { ElementConfig } from "../elements/types.js";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+  image: string;
+  elements: ElementConfig[];
+}
+
+class HuiPictureElementsCard extends LitElement implements LovelaceCard {
+  protected config?: Config;
+  private _hass?: HomeAssistant;
+
+  static get properties() {
+    return {
+      config: {},
+    };
+  }
+
+  set hass(hass: HomeAssistant) {
+    this._hass = hass;
+    for (const el of this.shadowRoot!.querySelectorAll("#root > *")) {
+      const element = el as LovelaceCard;
+      element.hass = this._hass;
+    }
+  }
+
+  public getCardSize(): number {
+    return 4;
+  }
+
+  public setConfig(config: Config): void {
+    if (!config) {
+      throw new Error("Invalid Configuration");
+    } else if (!config.image) {
+      throw new Error("Invalid Configuration: image required");
+    } else if (!Array.isArray(config.elements)) {
+      throw new Error("Invalid Configuration: elements required");
+    }
+
+    this.config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.config || !this.config.image || !this.config.elements) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this.config.title}">
+        <img src="${this.config.image}">
+        <div id="root">
+          ${this.config.elements.map((elementConfig: ElementConfig) =>
+            this._createHuiElement(elementConfig)
+          )}
+        </div>
+      </ha-card>
+    `;
+  }
+
+  private renderStyle(): TemplateResult {
+    return html`
+      <style>
+        ha-card {
+          overflow: hidden;
+        }
+        #root {
+          position: relative;
+          overflow: hidden;
+        }
+        #root img {
+          display: block;
+          width: 100%;
+        }
+        .element {
+          position: absolute;
+          transform: translate(-50%, -50%);
+        }
+      </style>
+    `;
+  }
+
+  private _createHuiElement(elementConfig: ElementConfig): LovelaceCard {
+    const element = createHuiElement(elementConfig) as LovelaceCard;
```

Comment:
These are not `LovelaceCard`, we need a new type for the elements.

Reply:
This raises a different consideration. As it is now, lovelace cards, entities and row elements have the same interface and can (by design or accident) be used interchangeably. This allows e.g. for adding an input_number slider on a picture-elements card via
```yaml
- type: custom:hui-input-number-entity-row
  entity: input_number.tod_sunrise_offset
  name: ' '
  icon: none
  style: ...
```

Should this compatibility be represented in the Interfaces?

Reply:
I think that we should not put this compatibility in our interfaces as it's something that works now by accident, but is not something I want to commit to supporting as it will put constraints on how the different types can evolve.

Reply:
I did consider creating such an interface but opted not to for the reasons @thomasloven stated. So we want a `LovelaceElement` interface?
```
export interface LovelaceElement extends HTMLElement {
  hass?: HomeAssistant;
  setConfig(config: LovelaceElementConfig): void;
}
```